core/blockchain.go

```diff
@@ -837,6 +845,13 @@ func (bc *BlockChain) writeHeadBlock(block *types.Block) {
 	}
 	bc.currentBlock.Store(block)
 	headBlockGauge.Update(int64(block.NumberU64()))
+
+	nextBlockNum := new(big.Int).Add(block.Number(), big.NewInt(1))
+	if bc.Config().IsL2Migration(nextBlockNum) {
+		log.Info("The next block is the L2 migration block, stopping block insertion", "currentBlock", block.NumberU64(), "hash", block.Hash(), "nextBlock", nextBlockNum.Uint64())
+		bc.StopInsert()
+		return
+	}
```

Comment:
I don't think we need this, the check up front should be sufficient. Sopping insertion isn't really important unless there is an attempt to insert the next block.

Reply:
When I remove this the long test actually starts failing. I was under the impression bc.StopInsert() was essential to call based on what I saw during testing. The function flips an atomic value that is checked at the beginning of numerous functions related to block insertion.